lib/Util/RequestOptions.php

```diff
@@ -86,27 +101,46 @@ public static function parse($options)
         }
 
         if (\is_string($options)) {
+            if ($strict) {
+                $message = 'Do not pass a string for request options. If you want to set the '
+                    . 'API key, pass an array like ["api_key" => <apiKey>] instead.';
+
+                throw new \Stripe\Exception\InvalidArgumentException($message);
+            }
+
             return new RequestOptions($options, [], null);
         }
 
         if (\is_array($options)) {
             $headers = [];
             $key = null;
             $base = null;
+
             if (\array_key_exists('api_key', $options)) {
                 $key = $options['api_key'];
+                unset($options['api_key']);
             }
             if (\array_key_exists('idempotency_key', $options)) {
                 $headers['Idempotency-Key'] = $options['idempotency_key'];
+                unset($options['idempotency_key']);
             }
             if (\array_key_exists('stripe_account', $options)) {
                 $headers['Stripe-Account'] = $options['stripe_account'];
+                unset($options['stripe_account']);
             }
             if (\array_key_exists('stripe_version', $options)) {
                 $headers['Stripe-Version'] = $options['stripe_version'];
+                unset($options['stripe_version']);
```

Comment:
can you add a clear test proving that `unset` can't mutate the input array? I know we discussed this and it's safe but that always makes me uneasy

Reply:
I've verified it manually, but I'd rather not add a test for it since it's how PHP itself works. We'd be testing the language, not the stripe-php library.